src/descriptor/template.rs

```diff
@@ -44,24 +45,48 @@ pub type DescriptorTemplateOut = (ExtendedDescriptor, KeyMap, ValidNetworks);
 /// Since [`ToWalletDescriptor`] is implemented for any [`DescriptorTemplate`], they can also be
 /// passed directly to the [`Wallet`](crate::Wallet) constructor.
 ///
-/// ## Example
+/// ## Examples
 ///
 /// ```
+/// use bdk::descriptor::error::Error as DescriptorError;
 /// use bdk::keys::{KeyError, ToDescriptorKey};
 /// use bdk::miniscript::Legacy;
 /// use bdk::template::{DescriptorTemplate, DescriptorTemplateOut};
 ///
 /// struct MyP2PKH<K: ToDescriptorKey<Legacy>>(K);
 ///
 /// impl<K: ToDescriptorKey<Legacy>> DescriptorTemplate for MyP2PKH<K> {
-///     fn build(self) -> Result<DescriptorTemplateOut, KeyError> {
+///     fn build(self) -> Result<DescriptorTemplateOut, DescriptorError> {
 ///         Ok(bdk::descriptor!(pkh(self.0))?)
 ///     }
 /// }
 /// ```
+///
+/// ```
+/// use bdk::keys::{KeyError, ToDescriptorKey};
+/// use bdk::miniscript::{Legacy, Segwitv0};
+/// use bdk::template::{DescriptorTemplate, DescriptorTemplateOut};
+/// use bdk::descriptor::error::Error as DescriptorError;
+///
+/// pub struct TimeDecayingMultisig<K> {
+///     pk_a: K,
+///     pk_b: K,
+///     timelock: u32,
+/// }
+///
+/// impl<K: ToDescriptorKey<Segwitv0>> DescriptorTemplate for TimeDecayingMultisig<K> {
+///     fn build(self) -> Result<DescriptorTemplateOut, DescriptorError> {
+///         let mut uncompressed_pk = bdk::bitcoin::PrivateKey::from_wif("L5EZftvrYaSudiozVRzTqLcHLNDoVn7H5HSfM9BAN6tMJX8oTWz6").unwrap();
+///         uncompressed_pk.compressed = false;
+///
+///         // Ok(bdk::descriptor!(wsh(thresh(2,pk(self.pk_a),s:pk(self.pk_b),s:d:v:older(self.timelock))))?)
```

Comment:
❓  Is this comment meant to be here? I am not sure what it's getting across.

Reply:
Nope, looks like I forgot to remove it...

Reply:
Actually, this new test I've added doesn't really make sense: I wanted to show that it correctly fails if you use uncompressed pks in a segwit script, but it's not a compile-time error, it's a runtime error. I would have to not only declare the template but also build a wallet and use it in there to actually show the failure..

Besides, I've also already added a test for that in `src/descriptor/dsl.rs`, so I think I'm just gonna remove this one